### PR TITLE
Update PR template

### PR DESCRIPTION
**Pull request by JDBetteridge**
So that it no longer looks like this:
```
---
name: "Bug fix or new feature"
description: ''
title: ''
labels: ''
assignees: ''

---

# Description
Please include a summary of the changes introduced by this PR.
Additionally include any relevant motivation and context. List any
dependencies that are required for this change, other PRs upon
which this may depend, and associated issues.

## Associated Pull Requests:
- Make a list (with links!)
- [Like this PR#691](https://github.com/OP2/PyOP2/pull/691)

## Fixes Issues:
- Make a list of issues (with links!)
- [Like this Issue#2824](https://github.com/firedrakeproject/firedrake/issues/2824)

If issues are fixed by this PR, prepend each of them with the word
"fixes", so they are automatically closed when this PR is merged. For
example "fixes #123, fixes #456".

# Checklist for author:

<!--
If you think an option is not relevant to your PR, do not delete it but use ~strikethrough formating on it~. This helps keeping track of the entire list.
-->

- [ ] I have linted the codebase (`make lint` in the `firedrake` source directory).
- [ ] My changes generate no new warnings.
- [ ] All of my functions and classes have appropriate docstrings.
- [ ] I have commented my code where its purpose may be unclear.
- [ ] I have included and updated any relevant documentation.
- [ ] Documentation builds locally (`make linkcheck; make html; make latexpdf` in `firedrake/docs` directory)
- [ ] I have added tests specific to the issues fixed in this PR.
- [ ] I have added tests that exercise the new functionality I have introduced
- [ ] Tests pass locally (`pytest tests` in the `firedrake` source directory) (useful, but not essential if you don't have suitable hardware).
- [ ] I have performed a self-review of my own code using the below guidelines.

# Checklist for reviewer:

- [ ] Docstrings present
- [ ] New tests present
- [ ] Code correctly commented
- [ ] No bad "code smells"
- [ ] No issues in parallel
- [ ] No CI issues (excessive parallelism/memory usage/time/warnings generated)
- [ ] Upstream/dependent branches and PRs are ready

Feel free to add reviewers if you know there is someone who is already aware of this work.

Please open this PR initially as a draft and mark as ready for review once CI tests are passing.

<!--
Thanks for contributing!
-->
```

### CODE REVIEW

**Comment by ReubenHill** [comment]
I thought the aim was just to get rid of the 
```
---
name: "Bug fix or new feature"
description: ''
title: ''
labels: ''
assignees: ''

---
```
bit?

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,61 +1,15 @@
----
-name: "Bug fix or new feature"
-description: ''
-title: ''
-labels: ''
-assignees: ''
-
----
-
 # Description
```
Comment (connorjward) [inline .github/pull_request_template.md:1]:
```suggestion
```
Is this line necessary?

**Review by connorjward — COMMENTED** [review]
I think putting all this into a comment is the right way to go. Perhaps the checklists should go into a policy ([example](https://github.com/firedrakeproject/firedrake/wiki/macOS-version-support))?

**Comment by JDBetteridge** [comment]
@ReubenHill it is! See the diff

**Comment by ReubenHill** [comment]
> @ReubenHill it is! See the diff

But the diff gets rid of all the tickboxes etc too

File: .github/pull_request_template.md (hunk 1 above)
Comment (JDBetteridge) [reply .github/pull_request_template.md:1]: I like the idea of having some sort of prompt. I'd really like to have it as a form, but this is not available on github

**Comment by JDBetteridge** [comment]
I know, but nobody was using them. My original intent was to add this so that essentially anyone could do a code review by following the steps and people who were unsure of what to do could just follow the steps.

I still like the idea, but if it's not being used we should probably get rid of it. A lot of PRs just deleted all the text anyway :shrug:

**Comment by ReubenHill** [comment]
> I know, but nobody was using them. My original intent was to add this so that essentially anyone could do a code review by following the steps and people who were unsure of what to do could just follow the steps.
> 
> I still like the idea, but if it's not being used we should probably get rid of it. A lot of PRs just deleted all the text anyway 🤷

I used it...

**Comment by JDBetteridge** [comment]
I know, and so did a couple of other people, but sadly you were in a minority. What I'd really like is a bunch of PR templates that you could select from like you can with issue templates, but it seems like this feature hasn't landed in github's interface yet. I have set up a folder with other templates for when they do enable it though!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,61 +1,15 @@
...
 
-- [ ] Docstrings present
-- [ ] New tests present
-- [ ] Code correctly commented
-- [ ] No bad "code smells"
-- [ ] No issues in parallel
-- [ ] No CI issues (excessive parallelism/memory usage/time/warnings generated)
-- [ ] Upstream/dependent branches and PRs are ready
+If issues are fixed by this PR, include link to them and prepend each of them with the word "fixes", so they are automatically closed when this PR is merged.
+For example "fixes #123, fixes #456"..
```
Comment (wence-) [inline .github/pull_request_template.md:9]: This (because no-one deletes the text) links every PR for all eternity to these two issues. `fixes #xyz` ?

Comment (dham) [inline .github/pull_request_template.md:9]:
```suggestion
For example "fixes #xyz, fixes #abc".
```

**Pull request merged**